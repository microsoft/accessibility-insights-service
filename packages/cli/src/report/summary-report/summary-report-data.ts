// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export interface ViolationCountMap {
    [key: string]: number;
}

export interface UrlToReportMap {
    [key: string]: string;
}

export interface SummaryReportData {
    violationCountByRuleMap: ViolationCountMap;
    failedUrlToReportMap: UrlToReportMap;
    passedUrlToReportMap: UrlToReportMap;
    unScannableUrls: string[];
}
