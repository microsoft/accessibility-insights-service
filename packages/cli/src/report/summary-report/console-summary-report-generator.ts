// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { injectable } from 'inversify';
import { table } from 'table';
import { SummaryReportData, ViolationCountMap } from './summary-report-data';
import { SummaryReportGenerator } from './summary-report-generator';

@injectable()
export class ConsoleSummaryReportGenerator implements SummaryReportGenerator {
    public generateReport(summaryReportData: SummaryReportData): string {
        return `\n\n${this.getSummaryMessage(summaryReportData.violationCountByRuleMap)}`;
    }

    private getSummaryMessage(violationCountByRuleMap: ViolationCountMap): string {
        if (Object.keys(violationCountByRuleMap).length === 0) {
            return 'Congratulations! No failed automated checks were found.';
        } else {
            return `Automated checks summary results: ${this.getSummaryTableString(violationCountByRuleMap)}`;
        }
    }

    private getSummaryTableString(violationCountByRuleMap: ViolationCountMap): string {
        const tableInput: string[][] = [];

        tableInput.push(['Failed Rule', 'Count']);

        for (const key of Object.keys(violationCountByRuleMap)) {
            tableInput.push([key, `${violationCountByRuleMap[key]}`]);
        }

        return `\n${table(tableInput)}\n`;
    }
}
