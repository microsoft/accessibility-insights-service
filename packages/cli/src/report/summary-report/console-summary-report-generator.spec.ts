// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import 'reflect-metadata';

import { ConsoleSummaryReportGenerator } from './console-summary-report-generator';
import { SummaryReportData } from './summary-report-data';

// tslint:disable: no-object-literal-type-assertion

describe(ConsoleSummaryReportGenerator, () => {
    let testSubject: ConsoleSummaryReportGenerator;

    beforeEach(() => {
        testSubject = new ConsoleSummaryReportGenerator();
    });

    test.each([
        [
            'has violations',
            {
                failedUrlToReportMap: { url1: 'file1' },
                passedUrlToReportMap: { url1: 'file1' },
                violationCountByRuleMap: {
                    rule1WithALongName11111111111111111: 10,
                    rule2: 3,
                },
                unscannableUrls: {},
            } as SummaryReportData,
        ],
        [
            'has no violations',
            {
                failedUrlToReportMap: { url1: 'file1' },
                passedUrlToReportMap: { url1: 'file1' },
                violationCountByRuleMap: {},
                unscannableUrls: {},
            } as SummaryReportData,
        ],
    ])('logs summary when %s', async (testCaseName, testCase) => {
        expect(testSubject.generateReport(testCase)).toMatchSnapshot();
    });
});
