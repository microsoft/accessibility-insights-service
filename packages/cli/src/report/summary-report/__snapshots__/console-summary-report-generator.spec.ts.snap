// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleSummaryReportGenerator logs summary when has no violations 1`] = `
"

Congratulations! No failed automated checks were found."
`;

exports[`ConsoleSummaryReportGenerator logs summary when has violations 1`] = `
"

Automated checks summary results: 
╔═════════════════════════════════════╤═══════╗
║ Failed Rule                         │ Count ║
╟─────────────────────────────────────┼───────╢
║ rule1WithALongName11111111111111111 │ 10    ║
╟─────────────────────────────────────┼───────╢
║ rule2                               │ 3     ║
╚═════════════════════════════════════╧═══════╝

"
`;
